### ISSUE TITLE
thread 'rustc' panicked at 'explicit panic', ../src/librbml/lib.rs:436

### ISSUE THREAD

**Issue by dwillie** [rust-lang/rust#33733]
I attempted to do a release build (`cargo build --release`) and received this message:

    Compiling mqtt-rust-sms v0.1.0 (file:///Users/dwillie/projects/detox/mqtt-rust-sms)
    ERROR:rbml::reader: failed to find block with tag 272
    error: internal compiler error: unexpected panic
    note: the compiler unexpectedly panicked. this is a bug.
    note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

So I'm here to report it. It's probably worth noting that `cargo build` and `cargo run` both function as expected. 

## Meta

`rustc --version --verbose`:

```
rustc 1.8.0 (db2939409 2016-04-11)
binary: rustc
commit-hash: db2939409db26ab4904372c82492cd3488e4c44e
commit-date: 2016-04-11
host: x86_64-apple-darwin
release: 1.8.0
```

Backtrace:

```
RUST_BACKTRACE=1 cargo build --release --verbose
       Fresh winapi-build v0.1.1
       Fresh pkg-config v0.3.8
       Fresh byteorder v0.4.2
       Fresh winapi v0.2.7
       Fresh typeable v0.1.2
       Fresh traitobject v0.0.1
       Fresh bitflags v0.7.0
       Fresh rustc-serialize v0.3.19
       Fresh mqtt3 v0.1.2
       Fresh gcc v0.3.28
       Fresh language-tags v0.2.2
       Fresh log v0.3.6
       Fresh lazy_static v0.2.1
       Fresh unicode-normalization v0.1.2
       Fresh semver v0.1.20
       Fresh hpack v0.2.0
       Fresh mime v0.2.0
       Fresh matches v0.1.2
       Fresh kernel32-sys v0.2.2
       Fresh libc v0.2.11
       Fresh rustc_version v0.1.7
       Fresh solicit v0.4.4
       Fresh httparse v1.1.2
       Fresh unicode-bidi v0.2.3
       Fresh num_cpus v0.2.12
       Fresh openssl-sys v0.7.12
       Fresh time v0.1.35
       Fresh rand v0.3.14
       Fresh idna v0.1.0 (https://github.com/servo/rust-url#d91d1751)
       Fresh idna v0.1.0
       Fresh openssl-sys-extras v0.7.12
       Fresh url v1.1.0 (https://github.com/servo/rust-url#d91d1751)
       Fresh url v1.1.0
       Fresh openssl v0.7.12
       Fresh unicase v1.4.0
       Fresh cookie v0.2.4
       Fresh openssl-verify v0.1.0
       Fresh netopt v0.1.2
       Fresh hyper v0.9.4
       Fresh mqttc v0.1.2
   Compiling mqtt-rust-sms v0.1.0 (file:///Users/dwillie/projects/detox/mqtt-rust-sms)
     Running `rustc src/main.rs --crate-name mqtt_rust_sms --crate-type bin -C opt-level=3 --out-dir /Users/dwillie/projects/detox/mqtt-rust-sms/target/release --emit=dep-info,link -L dependency=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release -L dependency=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps --extern url=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/liburl-6d05377aad30fd8a.rlib --extern netopt=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/libnetopt-6a3d95bef6af6a4c.rlib --extern hyper=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/libhyper-b69eb99d3b5f3595.rlib --extern rustc_serialize=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/librustc_serialize-d9e72695d437325f.rlib --extern mqttc=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/libmqttc-2a13e0acf9a90401.rlib --extern mqtt3=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/libmqtt3-f8a6d2aa515975b8.rlib -L native=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/build/openssl-4bcadbce25d39fc2/out -L native=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/build/openssl-sys-extras-e41b587668d68691/out -L native=/usr/local/opt/openssl/lib`
ERROR:rbml::reader: failed to find block with tag 272
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
thread 'rustc' panicked at 'explicit panic', ../src/librbml/lib.rs:436
stack backtrace:
   1:        0x107e26168 - sys::backtrace::tracing::imp::write::h3b28049ffa6d6406ZZu
   2:        0x107e2ead5 - panicking::default_handler::_$u7b$$u7b$closure$u7d$$u7d$::closure.43942
   3:        0x107e2e5d7 - panicking::default_handler::h9792cd244a79d9daorz
   4:        0x107df1d86 - sys_common::unwind::begin_unwind_inner::h127fbab41243a988LYt
   5:        0x107ce8484 - sys_common::unwind::begin_unwind::h7721608091404376439
   6:        0x107ce5ac4 - reader::get_doc::h900c8d0cd450eb95ykb
   7:        0x1043fe9d6 - creader::CrateReader<'a>::register_crate::h62645edecf14372cTZj
   8:        0x104400f58 - creader::CrateReader<'a>::resolve_crate::he1542772b02d0896E3j
   9:        0x104401d5d - creader::CrateReader<'a>::resolve_crate_deps::_$u7b$$u7b$closure$u7d$$u7d$::closure.38771
  10:        0x104401afb - collections::hash::map::HashMap<K, V, S>.FromIterator<(K, V)>::from_iter::h3969893598729083442
  11:        0x1043fde39 - creader::CrateReader<'a>::register_crate::h62645edecf14372cTZj
  12:        0x104400f58 - creader::CrateReader<'a>::resolve_crate::he1542772b02d0896E3j
  13:        0x104401d5d - creader::CrateReader<'a>::resolve_crate_deps::_$u7b$$u7b$closure$u7d$$u7d$::closure.38771
  14:        0x104401afb - collections::hash::map::HashMap<K, V, S>.FromIterator<(K, V)>::from_iter::h3969893598729083442
  15:        0x1043fde39 - creader::CrateReader<'a>::register_crate::h62645edecf14372cTZj
  16:        0x104400f58 - creader::CrateReader<'a>::resolve_crate::he1542772b02d0896E3j
  17:        0x1043fb1ac - creader::LocalCrateReader<'a, 'b>.Visitor<'hir>::visit_item::hd760834db1c6f482eDj
  18:        0x10440d7cd - creader::LocalCrateReader<'a, 'b>::read_crates::hb36235539ae0e4e04vk
  19:        0x103be0108 - driver::phase_3_run_analysis_passes::h5242075341164975033
  20:        0x103bb6c4d - driver::compile_input::h8dcd247802517a6fHca
  21:        0x103ba6308 - run_compiler::hb0408bcf47642fe6mPc
  22:        0x103ba3965 - sys_common::unwind::try::try_fn::h17587884860915543117
  23:        0x107e23b3b - __rust_try
  24:        0x107e1bd63 - sys_common::unwind::inner_try::ha8c5d452cd60c627NVt
  25:        0x103ba4219 - boxed::F.FnBox<A>::call_box::h11079460496650762403
  26:        0x107e2dacc - sys::thread::Thread::new::thread_start::h1c23c8220fd8bb7dPFy
  27:     0x7fff892c3c12 - _pthread_body
  28:     0x7fff892c3b8f - _pthread_start

Could not compile `mqtt-rust-sms`.

Caused by:
  Process didn't exit successfully: `rustc src/main.rs --crate-name mqtt_rust_sms --crate-type bin -C opt-level=3 --out-dir /Users/dwillie/projects/detox/mqtt-rust-sms/target/release --emit=dep-info,link -L dependency=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release -L dependency=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps --extern url=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/liburl-6d05377aad30fd8a.rlib --extern netopt=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/libnetopt-6a3d95bef6af6a4c.rlib --extern hyper=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/libhyper-b69eb99d3b5f3595.rlib --extern rustc_serialize=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/librustc_serialize-d9e72695d437325f.rlib --extern mqttc=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/libmqttc-2a13e0acf9a90401.rlib --extern mqtt3=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/deps/libmqtt3-f8a6d2aa515975b8.rlib -L native=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/build/openssl-4bcadbce25d39fc2/out -L native=/Users/dwillie/projects/detox/mqtt-rust-sms/target/release/build/openssl-sys-extras-e41b587668d68691/out -L native=/usr/local/opt/openssl/lib` (exit code: 101)
```

**Comment by dwillie** [comment]
Aaaand it's gone. 😂 

Seems I've been visited by the same ghost as Pierre! (https://github.com/rust-lang/rust/issues/33015)

...but I'm not drunk. I haven't made any code or project changes (no cleans etc.) that I can think of, or done any software updates. It just seems to have gone away on its own.

I did at one point attempt to use https://github.com/neil-lobracco/twilio-rs but it was having issues compiling and complaining about the `Sized` trait in a return type somewhere (I'm very new to Rust, sorry). 

This pull request would fix it apparently and mentioned v1.7.0 (https://github.com/neil-lobracco/twilio-rs/pull/6)

So this is maybe related to https://github.com/rust-lang/rust/issues/33292

Sorry if that's a bit scattered. Like I said, I'm new to Rust and so I'm just trying to connect some dots with what knowledge I do have. I promise I'm not drunk.

**Comment by adimarco** [comment]
I'm getting the same thing.  `cargo build` works fine, but `cargo test` explodes with the same errors you were seeing.

Rustc version info:
```
$ rustc --version --verbose
rustc 1.9.0 (e4e8b6668 2016-05-18)
binary: rustc
commit-hash: e4e8b666850a763fdf1c3c2c142856ab51e32779
commit-date: 2016-05-18
host: x86_64-unknown-linux-gnu
release: 1.9.0
```

And the error output:
```
$ RUST_BACKTRACE=1 cargo test --features all
   Compiling rusoto v0.14.1 (file:///home/adimarco/code/rusoto)
ERROR:rbml::reader: failed to find block with tag 272
ERROR:rbml::reader: failed to find block with tag 272
ERROR:rbml::reader: failed to find block with tag 272
ERROR:rbml::reader: failed to find block with tag 272
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'explicit panic', ../src/librbml/lib.rs:436
stack backtrace:
   1:     0x7f165f6a4570 - std::sys::backtrace::tracing::imp::write::h4c73fcd3363076f5
   2:     0x7f165f6b1c6b - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::h0422dbb3077e6747
   3:     0x7f165f6b180c - std::panicking::default_hook::haac48fa641db8fa2
   4:     0x7f165f67679f - std::sys_common::unwind::begin_unwind_inner::h39d40f52add53ef7
   5:     0x7f1656c2ac3f - std::sys_common::unwind::begin_unwind::hd78480d5ec5c51b2
   6:     0x7f1656c28906 - rbml::reader::get_doc::hdb38d8b870a9741e
   7:     0x7f165d49282e - rustc_metadata::creader::CrateReader::register_crate::hdfd966766b932c68
   8:     0x7f165d497562 - rustc_metadata::creader::CrateReader::read_extension_crate::hb0997de4fb0918fb
   9:     0x7f165d49985e - rustc_metadata::creader::CrateReader::read_exported_macros::hf3b61431145aafb6
  10:     0x7f165d4bd0e8 - _<macro_import..MacroLoader<'a> as syntax..visit..Visitor<'v>>::visit_item::h668e7bfc1acab9b3
  11:     0x7f165d4bbeea - rustc_metadata::macro_import::read_macro_defs::h240a20dad60749cc
  12:     0x7f165fbda9ec - rustc_driver::driver::phase_2_configure_and_expand::h505e157de342d54f
  13:     0x7f165fbc109a - rustc_driver::driver::compile_input::h6491aaddd9e61258
  14:     0x7f165fba94e4 - rustc_driver::run_compiler::h80b2ba5e4d787c5f
  15:     0x7f165fba6941 - std::sys_common::unwind::try::try_fn::h34e27823ddd1d5e9
  16:     0x7f165f6a1d0b - __rust_try
  17:     0x7f165f6a1c9d - std::sys_common::unwind::inner_try::h9eebd8dc83f388a6
  18:     0x7f165fba718a - _<F as std..boxed..FnBox<A>>::call_box::h3d5d78986dfac5b2
  19:     0x7f165f6afe04 - std::sys::thread::Thread::new::thread_start::h471ad90789353b5b
  20:     0x7f165806e6f9 - start_thread
  21:     0x7f165f305b5c - clone
  22:                0x0 - <unknown>

Build failed, waiting for other jobs to finish...
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'explicit panic', ../src/librbml/lib.rs:436
stack backtrace:
   1:     0x7f5eab7bf570 - std::sys::backtrace::tracing::imp::write::h4c73fcd3363076f5
   2:     0x7f5eab7ccc6b - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::h0422dbb3077e6747
   3:     0x7f5eab7cc80c - std::panicking::default_hook::haac48fa641db8fa2
   4:     0x7f5eab79179f - std::sys_common::unwind::begin_unwind_inner::h39d40f52add53ef7
   5:     0x7f5ea2d45c3f - std::sys_common::unwind::begin_unwind::hd78480d5ec5c51b2
   6:     0x7f5ea2d43906 - rbml::reader::get_doc::hdb38d8b870a9741e
   7:     0x7f5ea95ad82e - rustc_metadata::creader::CrateReader::register_crate::hdfd966766b932c68
   8:     0x7f5ea95b079f - rustc_metadata::creader::CrateReader::resolve_crate::h98decfaedbcfe12a
   9:     0x7f5ea95a9b41 - _<creader..LocalCrateReader<'a, 'b> as rustc..hir..intravisit..Visitor<'hir>>::visit_item::h2f4b84ef1536be92
  10:     0x7f5ea95be4be - rustc_metadata::creader::LocalCrateReader::read_crates::h7c2989bf6ef8974b
  11:     0x7f5eabd08c86 - rustc_driver::driver::phase_3_run_analysis_passes::h83da042ec4b8ea10
  12:     0x7f5eabcddf9f - rustc_driver::driver::compile_input::h6491aaddd9e61258
  13:     0x7f5eabcc44e4 - rustc_driver::run_compiler::h80b2ba5e4d787c5f
  14:     0x7f5eabcc1941 - std::sys_common::unwind::try::try_fn::h34e27823ddd1d5e9
  15:     0x7f5eab7bcd0b - __rust_try
  16:     0x7f5eab7bcc9d - std::sys_common::unwind::inner_try::h9eebd8dc83f388a6
  17:     0x7f5eabcc218a - _<F as std..boxed..FnBox<A>>::call_box::h3d5d78986dfac5b2
  18:     0x7f5eab7cae04 - std::sys::thread::Thread::new::thread_start::h471ad90789353b5b
  19:     0x7f5ea41896f9 - start_thread
  20:     0x7f5eab420b5c - clone
  21:                0x0 - <unknown>

error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'explicit panic', ../src/librbml/lib.rs:436
stack backtrace:
   1:     0x7f1f53753570 - std::sys::backtrace::tracing::imp::write::h4c73fcd3363076f5
   2:     0x7f1f53760c6b - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::h0422dbb3077e6747
   3:     0x7f1f5376080c - std::panicking::default_hook::haac48fa641db8fa2
   4:     0x7f1f5372579f - std::sys_common::unwind::begin_unwind_inner::h39d40f52add53ef7
   5:     0x7f1f4acd9c3f - std::sys_common::unwind::begin_unwind::hd78480d5ec5c51b2
   6:     0x7f1f4acd7906 - rbml::reader::get_doc::hdb38d8b870a9741e
   7:     0x7f1f5154182e - rustc_metadata::creader::CrateReader::register_crate::hdfd966766b932c68
   8:     0x7f1f5154479f - rustc_metadata::creader::CrateReader::resolve_crate::h98decfaedbcfe12a
   9:     0x7f1f5153db41 - _<creader..LocalCrateReader<'a, 'b> as rustc..hir..intravisit..Visitor<'hir>>::visit_item::h2f4b84ef1536be92
  10:     0x7f1f515524be - rustc_metadata::creader::LocalCrateReader::read_crates::h7c2989bf6ef8974b
  11:     0x7f1f53c9cc86 - rustc_driver::driver::phase_3_run_analysis_passes::h83da042ec4b8ea10
  12:     0x7f1f53c71f9f - rustc_driver::driver::compile_input::h6491aaddd9e61258
  13:     0x7f1f53c584e4 - rustc_driver::run_compiler::h80b2ba5e4d787c5f
  14:     0x7f1f53c55941 - std::sys_common::unwind::try::try_fn::h34e27823ddd1d5e9
  15:     0x7f1f53750d0b - __rust_try
  16:     0x7f1f53750c9d - std::sys_common::unwind::inner_try::h9eebd8dc83f388a6
  17:     0x7f1f53c5618a - _<F as std..boxed..FnBox<A>>::call_box::h3d5d78986dfac5b2
  18:     0x7f1f5375ee04 - std::sys::thread::Thread::new::thread_start::h471ad90789353b5b
  19:     0x7f1f4c11d6f9 - start_thread
  20:     0x7f1f533b4b5c - clone
  21:                0x0 - <unknown>

error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'explicit panic', ../src/librbml/lib.rs:436
stack backtrace:
   1:     0x7f2df6386570 - std::sys::backtrace::tracing::imp::write::h4c73fcd3363076f5
   2:     0x7f2df6393c6b - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::h0422dbb3077e6747
   3:     0x7f2df639380c - std::panicking::default_hook::haac48fa641db8fa2
   4:     0x7f2df635879f - std::sys_common::unwind::begin_unwind_inner::h39d40f52add53ef7
   5:     0x7f2ded90cc3f - std::sys_common::unwind::begin_unwind::hd78480d5ec5c51b2
   6:     0x7f2ded90a906 - rbml::reader::get_doc::hdb38d8b870a9741e
   7:     0x7f2df417482e - rustc_metadata::creader::CrateReader::register_crate::hdfd966766b932c68
   8:     0x7f2df417779f - rustc_metadata::creader::CrateReader::resolve_crate::h98decfaedbcfe12a
   9:     0x7f2df4170b41 - _<creader..LocalCrateReader<'a, 'b> as rustc..hir..intravisit..Visitor<'hir>>::visit_item::h2f4b84ef1536be92
  10:     0x7f2df41854be - rustc_metadata::creader::LocalCrateReader::read_crates::h7c2989bf6ef8974b
  11:     0x7f2df68cfc86 - rustc_driver::driver::phase_3_run_analysis_passes::h83da042ec4b8ea10
  12:     0x7f2df68a4f9f - rustc_driver::driver::compile_input::h6491aaddd9e61258
  13:     0x7f2df688b4e4 - rustc_driver::run_compiler::h80b2ba5e4d787c5f
  14:     0x7f2df6888941 - std::sys_common::unwind::try::try_fn::h34e27823ddd1d5e9
  15:     0x7f2df6383d0b - __rust_try
  16:     0x7f2df6383c9d - std::sys_common::unwind::inner_try::h9eebd8dc83f388a6
  17:     0x7f2df688918a - _<F as std..boxed..FnBox<A>>::call_box::h3d5d78986dfac5b2
  18:     0x7f2df6391e04 - std::sys::thread::Thread::new::thread_start::h471ad90789353b5b
  19:     0x7f2deed506f9 - start_thread
  20:     0x7f2df5fe7b5c - clone
  21:                0x0 - <unknown>
```
